nengo_fpga/networks/fpga_pes_ensemble_network.py

```diff
@@ -143,18 +141,43 @@ def cleanup(self):
             os.remove(self.local_data_filepath)
 
     def connect_thread_func(self):
+        # Get the IP of the remote device from the fpga_config file
         remote_ip = fpga_config.get(self.fpga_name, 'ip')
 
+        # Get the SSH options from the fpga_config file
+        ssh_port = fpga_config.get(self.fpga_name, 'ssh_port')
+        ssh_user = fpga_config.get(self.fpga_name, 'ssh_user')
+
+        if fpga_config.has_option(self.fpga_name, 'ssh_pwd'):
+            ssh_pwd = fpga_config.get(self.fpga_name, 'ssh_pwd')
+        else:
+            ssh_pwd = None
+
+        if fpga_config.has_option(self.fpga_name, 'ssh_key'):
+            ssh_key = os.path.expanduser(fpga_config.get(self.fpga_name,
+                                                         'ssh_key'))
+        else:
+            ssh_key = None
+
         # Connect to remote location over ssh
-        self.ssh_client.connect(
-            fpga_config.get(self.fpga_name, 'ip'),
-            port=fpga_config.get(self.fpga_name, 'port'),
-            username=fpga_config.get(self.fpga_name, 'user'),
-            password=fpga_config.get(self.fpga_name, 'pwd'))
+        if ssh_key is not None:
+            # If an ssh key is provided, just use it
+            self.ssh_client.connect(remote_ip, port=ssh_port,
+                                    username=ssh_user, key_filename=ssh_key)
+        elif ssh_pwd is not None:
+            # If an ssh password is provided, just use it
+            self.ssh_client.connect(remote_ip, port=ssh_port,
+                                    username=ssh_user, password=ssh_pwd)
+        else:
+            # If no password or key is specified, just use the default connect
+            # (paramiko will then try to connect using the id_rsa file in the
+            #  ~/.ssh/ folder)
```

Comment:
If this is the default behaviour do we need this in `fpga_config`?

Reply:
Yes. The entry in `fpga_config` allows you to specify a key that is different from the one stored in the default location. This is more useful in windows, but I can see users doing it in linux/mac as well.